src/test/java/core/JenkinsDatabaseSecurityRealmTest.java

```diff
@@ -67,7 +68,14 @@ public void login_and_logout() {
 
         jenkins.logout();
 
-        assertEquals(null, jenkins.getCurrentUser());
+        try {
+            jenkins.getCurrentUser().id();
+            // id() now calls load() which will
+            // attempt to getJson() since id == null
+            // for logged-out user.
+            fail("id() Should throw RuntimeException for logged-out user");
+        } catch (RuntimeException e) {
```

Comment:
I believe we should fix `User.getCurrent()` to preserve the null semantics. The one I broke in c26b4e0cec687f5d802c6d4805bf6c09c5ea6790.